compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2444,6 +2446,63 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         }
     }
 
+    fn undeclared_module_suggest_declare(
+        &mut self,
+        ident: Ident,
+        path: std::path::PathBuf,
+    ) -> Option<(Vec<(Span, String)>, String, Applicability)> {
+        let map = self.tcx.sess.source_map();
+        let sp = map
+            .span_extend_to_prev_char(
+                map.span_extend_to_prev_char(ident.span, '\n', true),
```

Comment:
Could you add a comment explaining what are you trying to do here?
(Also format this on a separate line to improve readability `let sp = map.span_extend_to_prev_char(ident.span, '\n', true);`)

Reply:
Ah, wait, I think this can actually use `current_crate_outer_attr_insert_span` instead of `UsePlacementFinder`.
(The newly added comment tells that the current logic is clearly incorrect.)